### PR TITLE
[12.0][ADD] purchase_order_line_price_history: new module

### PR DESCRIPTION
**Pull request by ernestotejeda**
similar to https://github.com/OCA/sale-workflow/tree/12.0/sale_order_line_price_history, but for purchases.

TT24733

### CODE REVIEW

File: purchase_order_line_price_history/README.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+To use this module, you need to:
+
+#. Go to *Purchase -> Requests for Quotation* and select
+   a Requests for Quotation or create a new one.
+#. Click on the new clock button in one of the order lines.
+#. A pop-up will open and you will see the *price history* for the product of
+   the order line and for the customer of the selected or created
+   Requests for Quotation.
+#. You can select other customer or leave it empty to see the price history for
```
Comment (sergio-teruel) [inline purchase_order_line_price_history/README.rst]: Change to vendor or supplier

File: purchase_order_line_price_history/README.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+   all customers.
```
Comment (sergio-teruel) [inline purchase_order_line_price_history/README.rst]: the same

File: purchase_order_line_price_history/i18n/es.po [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+#. module: purchase_order_line_price_history
+#: model:ir.model.fields,field_description:purchase_order_line_price_history.field_purchase_order_line_price_history__create_date
+#: model:ir.model.fields,field_description:purchase_order_line_price_history.field_purchase_order_line_price_history_line__create_date
+msgid "Created on"
+msgstr "Creado el"
+
+#. module: purchase_order_line_price_history
+#: model:ir.model.fields,field_description:purchase_order_line_price_history.field_purchase_order_line_price_history__partner_id
+msgid "Customer"
```
Comment (sergio-teruel) [inline purchase_order_line_price_history/i18n/es.po]: Reference to supplier

File: purchase_order_line_price_history/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+msgstr "Empresa"
+
+#. module: purchase_order_line_price_history
+#: model_terms:ir.ui.view,arch_db:purchase_order_line_price_history.purchase_order_form
+msgid "Price history"
+msgstr "Historial de compras"
+
+#. module: purchase_order_line_price_history
+#: model:ir.actions.act_window,name:purchase_order_line_price_history.purchase_order_line_price_history_action
+msgid "Price history per customer"
```
Comment (sergio-teruel) [inline purchase_order_line_price_history/i18n/es.po]: Reference to supplier

File: purchase_order_line_price_history/wizards/purchase_order_line_price_history.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        comodel_name='purchase.order.line',
+        string='Purchase order line',
+        default=lambda self: self.env.context.get("active_id"),
+    )
+    product_id = fields.Many2one(
+        related="purchase_order_line_id.product_id",
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Customer',
```
Comment (sergio-teruel) [inline purchase_order_line_price_history/wizards/purchase_order_line_price_history.py]: Change by Supplier

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Comment by ernestotejeda** [comment]
@sergio-teruel Changes done.

**Review by sergio-teruel — APPROVED** [review]

File: purchase_order_line_price_history/tests/test_purchase_order_line_price_history.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+from odoo.tests import Form, SavepointCase
+
+
+class TestPurchaseOrderLinePriceHistory(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner_1 = cls.env.ref('base.res_partner_1')
```
Comment (pedrobaeza) [inline purchase_order_line_price_history/tests/test_purchase_order_line_price_history.py]: Don't use demo data, but create new partners and products.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**